common/tdx/quote.c

```diff
@@ -0,0 +1,299 @@
+        OE_RAISE_MSG(
+            OE_REPORT_PARSE_ERROR,
+            "Unexpected quote length while parsing TDX quote.",
+            NULL);
+
+    p += sizeof(tdx_qe_report_certification_data_t);
+
+    qe_auth_data->size = _read_uint_16(p);
+    p += 2;
+    qe_auth_data->data = (uint8_t*)p;
+    p += qe_auth_data->size;
+
+    if (p > quote_end)
+        OE_RAISE_MSG(
+            OE_REPORT_PARSE_ERROR,
+            "Parse error after parsing QE authorization data.",
+            NULL);
+
+    qe_cert_data->type = _read_uint_16(p);
+    p += 2;
+    qe_cert_data->size = _read_uint_32(p);
+    p += 4;
+    qe_cert_data->data = (uint8_t*)p;
+    p += qe_cert_data->size;
+
+    if (p != quote_end)
+        OE_RAISE_MSG(
+            OE_REPORT_PARSE_ERROR,
+            "Unexpected quote length while parsing.",
+            NULL);
+
+    //
+    // Validation
+    //
+    OE_CHECK_MSG(
+        _validate_tdx_quote(tdx_quote), "SGX quote validation failed.", NULL);
+
+    OE_CHECK_MSG(
+        oe_validate_qe_cert_data(qe_cert_data),
+        "Failed to validate QE certificate data.",
+        NULL);
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+oe_result_t oe_verify_quote_with_tdx_endorsements(
+    const uint8_t* quote,
+    size_t quote_size,
+    const oe_sgx_endorsements_t* sgx_endorsements,
+    oe_datetime_t* input_validation_time,
+    oe_tcb_info_tcb_level_t* platform_tcb_level,
+    oe_datetime_t* valid_from,
+    oe_datetime_t* valid_until)
+{
+    oe_result_t result = OE_UNEXPECTED;
+
+    OE_TRACE_INFO("oe_verify_quote_with_tdx_endorsements");
+
+    oe_result_t get_sgx_quote_validity_result = OE_UNEXPECTED;
+    oe_datetime_t validity_from = {0};
+    oe_datetime_t validity_until = {0};
+    oe_datetime_t validation_time = {0};
+
+    uint32_t collateral_expiration_status;
+    uint32_t quote_verification_result;
+    uint8_t supplemental_data[MAX_SUPPLEMENTAL_DATA_SIZE] = {0};
+    uint32_t supplemental_data_size_out = 0;
+    time_t expiration_check_date = 0;
+
+    // Verify quote/endorsements for the given time. Use endorsements
+    // creation time if one was not provided.
+    if (input_validation_time == NULL)
+    {
+        OE_CHECK_MSG(
+            oe_datetime_from_string(
+                (const char*)(sgx_endorsements
+                                  ->items
+                                      [OE_SGX_ENDORSEMENT_FIELD_CREATION_DATETIME]
+                                  .data),
+                sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CREATION_DATETIME]
+                    .size,
+                &validation_time),
+            "Invalid creation time in endorsements: %s",
+            sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_CREATION_DATETIME]
+                .data);
+    }
+    else
+    {
+        validation_time = *input_validation_time;
+        // Use untrusted time if we cannot get time from inputs
+        // OE_CHECK(oe_datetime_now(&validation_time));
```

Comment:
Is this OE_CHECK meant to be commented out?

Reply:
Good catch. Forgot to clean them up.